integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/preaaggregate/PreAggregateUtil.scala

```diff
@@ -445,11 +443,11 @@ object PreAggregateUtil {
         .fromWrapperToExternalTableInfo(wrapperTableInfo, dbName, tableName)
       updateSchemaInfo(carbonTable,
         thriftTable)(sparkSession)
-      LOGGER.info(s"Parent table updated is successful for table $dbName.$tableName")
+      LOGGER.info(s"Parent table updated is successful for table" +
+                  s" $dbName.${childSchema.getRelationIdentifier.toString}")
     } catch {
       case e: Exception =>
         LOGGER.error(e, "Pre Aggregate Parent table update failed reverting changes")
-        revertMainTableChanges(dbName, tableName, numberOfCurrentChild)(sparkSession)
```

Comment:
Now where you are reverting the schema if updation fails?

Reply:
If updation of the parent table fails then the aggregate table will be dropped from undoMetadata. There is no need to revert the main table changes here.